src/main/java/com/arcbees/bourseje/server/upload/CloudStorageUploadService.java

```diff
@@ -0,0 +1,68 @@
+package com.arcbees.bourseje.server.upload;
+
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.tools.cloudstorage.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+
+public class CloudStorageUploadService implements ImageUploadService {
+    private static final int BUFFER_SIZE = 2 * 1024 * 1024;
+    private static final String BUCKET_NAME = "myBucket";
+
+    private final GcsService gcsService = GcsServiceFactory.createGcsService(new RetryParams.Builder()
+            .initialRetryDelayMillis(10)
+            .retryMaxAttempts(10)
+            .totalRetryPeriodMillis(15000)
+            .build());
+
+    @Override
+    public String upload(String name, InputStream inputStream, long size) {
+        GcsOutputChannel outputChannel;
+
+        GcsFileOptions options = new GcsFileOptions.Builder()
+                .mimeType("image/jpeg")
+                .acl("full-control") //TODO: Check if this is the right access control type.
+                .build();
+        GcsFilename fileName = getFileName(name);
+
+        try {
+            outputChannel = gcsService.createOrReplace(fileName, options);
+            copy(inputStream, Channels.newOutputStream(outputChannel));
```

Comment:
If you use apache commons-io, you could replace this with `IOUtils.copy(inputStream, Channels.newOutputStream(outputChannel))`